src/main/java/de/tum/in/www1/artemis/repository/ParticipationRepository.java

```diff
@@ -53,14 +53,14 @@
     @Query("select distinct participation from Participation participation left join fetch participation.results par where participation.exercise.id = :#{#exerciseId} and participation.student.id = :#{#studentId} and (par.id = (select max(id) from participation.results) or par.id = null)")
     Optional<Participation> findByExerciseIdAndStudentIdWithLatestResult(@Param("exerciseId") Long exerciseId, @Param("studentId") Long studentId);
 
-    @Query("select distinct participation from Participation participation left join fetch participation.submissions submission left join fetch submission.result result where participation.exercise.id = :#{#exerciseId} and submission.submitted = true and result is null")
-    List<Participation> findByExerciseIdWithEagerSubmittedSubmissionsWithoutResults(@Param("exerciseId") Long exerciseId);
+    @Query("select distinct participation from Participation participation left join fetch participation.submissions submission left join fetch submission.result result where participation.exercise.id = :#{#exerciseId} and submission.submitted = true and (result is null or result.assessmentType = 'AUTOMATIC')")
+    List<Participation> findByExerciseIdWithEagerSubmittedSubmissionsWithoutManualResults(@Param("exerciseId") Long exerciseId);
 
     @Query("select distinct participation from Participation participation left join fetch participation.results where participation.id = :#{#participationId}")
     Optional<Participation> findByIdWithEagerResults(@Param("participationId") Long participationId);
 
-    @Query("select distinct participation from Participation participation left join fetch participation.submissions where participation.id = :#{#participationId}")
-    Optional<Participation> findByIdWithEagerSubmissions(@Param("participationId") Long participationId);
+    @Query("select distinct participation from Participation participation left join fetch participation.submissions s left join fetch s.result left join fetch participation.results where participation.id = :#{#participationId}")
+    Optional<Participation> findByIdWithEagerSubmissionsAndResults(@Param("participationId") Long participationId);
```

Comment:
did you double check that other exercises than modeling exercises are not affected by this change?
Please add some JavaDoc here as well

Reply:
Yes, the corresponding method `findOneWithEagerSubmissionsAndResults` of the `ParticipationService` is only used by `getSubmissionForModelingEditor` in the `ModelingSubmissionResource`.

I added some documentation.